packages/api-providers/src/polkadot/vanchor-deposit.ts

```diff
@@ -3,31 +3,221 @@
+  const cachedURI = getCachedFixtureURI('proving_key_uncompressed_sub_vanchor_2_2_2.bin');
+  const ipfsKeyRequest = await fetch(withLocalFixtures() ? cachedURI : IPFSUrl);
+  const circuitKeyArrayBuffer = await ipfsKeyRequest.arrayBuffer();
+
+  logger.info('Done Fetching key');
+  const circuitKey = new Uint8Array(circuitKeyArrayBuffer);
+
+  return circuitKey;
+}
+
 export class PolkadotVAnchorDeposit extends VAnchorDeposit<WebbPolkadot, DepositPayload> {
-  generateBridgeNote(
-    anchorId: string | number,
+  private async getleafIndex(leaf: Uint8Array, indexBeforeInsertion: number, treeId: number): Promise<number> {
+    return getLeafIndex(this.inner.api, leaf, indexBeforeInsertion, treeId);
+  }
+
+  async generateBridgeNote(
+    _vanchorId: string | number, // always Zero as there will be only one vanchor
     destination: number,
     amount: number,
     wrappableAssetAddress?: string
   ): Promise<DepositPayload> {
-    throw new Error('Method not implemented.');
+    // Get the currency bridge currency
+    const currency = this.bridgeApi.currency;
+
+    // No currency is selected on the API
+    if (!currency) {
+      logger.error('Not currency/active bridge available');
+      throw new Error('api not ready');
+    }
+    const tokenSymbol = currency.view.symbol;
+    const destChainId = destination;
+    // Chain id of the active API
+    const chainId = await this.inner.api.consts.linkableTreeBn254.chainIdentifier;
+    const chainType = await this.inner.api.consts.linkableTreeBn254.chainType;
+    const sourceChainId = computeChainIdType(Number(chainType.toHex()), Number(chainId));
+    const anchors = await this.bridgeApi.getVariableAnchors();
+    const anchor = anchors[0];
+    // Tree id for the target chain
+    const treeId = anchor.neighbours[InternalChainId.ProtocolSubstrateStandalone] as number;
+
+    const noteInput: NoteGenInput = {
+      amount: currencyToUnitI128(amount).toString(),
+      backend: 'Arkworks',
+      curve: 'Bn254',
+      denomination: '18',
+      exponentiation: '5',
+      hashFunction: 'Poseidon',
+      protocol: 'vanchor',
+      sourceChain: sourceChainId.toString(),
+      sourceIdentifyingData: treeId.toString(),
+      targetChain: destChainId.toString(),
+      targetIdentifyingData: treeId.toString(),
+      tokenSymbol: tokenSymbol,
+      width: '5',
+      index: 0,
+    };
+
+    const note = await Note.generateNote(noteInput);
+    return {
+      note,
+      params: [treeId],
+    };
   }
 
-  deposit(depositPayload: DepositPayload): Promise<void> {
-    throw new Error('Method not implemented.');
+  async deposit(depositPayload: DepositPayload): Promise<VAnchorDepositResults> {
+    try {
+      // Getting the active account
+      const account = await this.inner.accounts.activeOrDefault;
+      const secret = randomAsU8a();
+      this.emit('stateChange', WithdrawState.GeneratingZk);
```

Comment:
As fetching fixtures is costly to some degree, and there are somewhere that this can error out can we set the fixtures just before the generation of zk ?

Reply:
I agree it is costly and likely to error, that is why I prefer it to be the first step. No sense in fetching leaves if we can't do anything with them due to an issue fetching fixtures.

Reply:
I Agree with that but it should be after checking for balance ...etc checks that aren't costly